Sources/AppCoinsSDK/UI/Purchase/ViewModels/AdyenViewModel.swift

```diff
@@ -65,8 +73,8 @@ internal class AdyenViewModel : ObservableObject {
         }
     }
 
-    internal func adyenFailedHandler() {
-        bottomSheetViewModel.transactionFailedWith(error: .systemError)
+    internal func adyenFailedHandler(description: String) {
+        self.bottomSheetViewModel.transactionFailedWith(error: .systemError(message: "adyenFailedHandler", description: description))
```

Comment:
"adyenFailedHandler" is not an appropriate error message, be descriptive of what is wrong and clear on how we should look for the problem.